proposals/nullable-reference-types-specification.md

```diff
@@ -0,0 +1,266 @@
+    ;
+
+warning_action
+    : ...
+    | 'enable'
+    ;
+```
+
+Note that the new form of `pragma_warning_body` uses `nullable_action`, not `warning_action`.
+
+
+# Nullable contexts
+
+Every line of source code has a *nullable annotation context* and a *nullable warning context*. These control whether nullable annotations have effect, and whether nullability warnings are given. The annotation context of a given line is either *disabled* or *enabled*. The warning context of a given line is either *disabled*, *safeonly* or *enabled*.
+
+Both contexts can be specified at the project level (outside of C# source code), or anywhere within a source file via `#nullable` and `#pragma warning` pre-processor directives. If no project level settings are provided the default is for both contexts to be *disabled*.
+
+The `#nullable` directive controls both the annotation and warning contexts within the source text, and take precedence over the project-level settings. The `#pragma warning ... nullable` directives control only the warning context, leaving the annotation context unchanged.
+
+A directive sets the context(s) it controls for subsequent lines of code, until another directive overrides it, or until the end of the source file.
+
+The effect of the directives is as follows:
+
+- `#nullable disable`: Sets the nullable annotation and warning contexts to *disabled*
+- `#nullable enable`: Sets the nullable annotation and warning contexts to *enabled*
+- `#nullable restore`: Restores the nullable annotation and warning contexts to project settings
+- `#nullable safeonly`: Sets the nullable annotation context to *enabled* and the warning context to *safeonly*
+- `#pragma warning disable nullable`: Sets the nullable warning context to *disabled*
+- `#pragma warning enable nullable`: Sets the nullable warning context to *enabled*
+- `#pragma warning restore nullable`: Restores the nullable warning context to project settings
+- `#pragma warning safeonly nullable`: Sets the nullable warning context to *safeonly*
+
+
+# Nullability of types
+
+A `type` is deemed to occur in a given annotation context when the last token of the type is within that context.
+
+A given type can have one of four nullabilities: *Oblivious*, *nonnullable*, *nullable* and *unknown*. *Nonnullable* and *unknown* types may yield warnings if a potential `null` value is assigned to them. Values of *nullable* and *unknown* types may yield warnings when dereferenced without proper null checking.
+
+The kind of type and the nullable annotation context it occurs in determine its nullability:
+
+- A nonnullable value type `S` is always *nonnullable*
+- A nullable value type `S?` is always *nullable*
+- An unannotated reference type `C` in a *disabled* annotation context is *oblivious*
+- An unannotated reference type `C` in an *enabled* annotation context is *nonnullable*
+- A nullable reference type `C?` in a *disabled* annotation context is *nullable* (but a warning is yielded)
```

Comment:
the warning here seems unfortunate.  if i'm in the middle of updating things, but decide to disable things because it's turning out to be a lot of work, would i then get warning?  Or is there a recommendation on what combination of pp directives i should use in this case to allow me to suspend my work?

Reply:
Would `git stash` be a good enough mitigation for this scenario?